source/Ldap.Integration.Tests/scripts/ActiveDirectory/Azure/main.tf

```diff
@@ -0,0 +1,170 @@
+terraform {
+  required_version = ">=0.12"
+  required_providers {
+    azurerm = {
+      source = "hashicorp/azurerm"
+      version = "~>2.0"
+    }
+  }
+}
+
+provider "azurerm" {
+  features {}
+}
+
+
+## <https://www.terraform.io/docs/providers/azurerm/r/resource_group.html>
+resource "azurerm_resource_group" "rg" {
+  name     = "rg_authtest_ad"
+  location =  var.resource_group_location
+}
+
+## <https://www.terraform.io/docs/providers/azurerm/r/virtual_network.html>
+resource "azurerm_virtual_network" "vnet" {
+  name                = "vNet1"
+  address_space       = ["10.0.0.0/16"]
+  location            = azurerm_resource_group.rg.location
+  resource_group_name = azurerm_resource_group.rg.name
+}
+
+## <https://www.terraform.io/docs/providers/azurerm/r/subnet.html> 
+resource "azurerm_subnet" "subnet" {
+  name                 = "subnet"
+  resource_group_name  = azurerm_resource_group.rg.name
+  virtual_network_name = azurerm_virtual_network.vnet.name
+  address_prefixes       = ["10.0.2.0/24"]
+}
+
+resource "azurerm_public_ip" "public_ip" {
+  name                = "public_ip"
+  resource_group_name = azurerm_resource_group.rg.name
+  location            = azurerm_resource_group.rg.location
+  allocation_method   = "Dynamic"
+}
+
+## <https://www.terraform.io/docs/providers/azurerm/r/network_interface.html>
+resource "azurerm_network_interface" "example" {
+  name                = "nic1"
+  location            = azurerm_resource_group.rg.location
+  resource_group_name = azurerm_resource_group.rg.name
+
+
+  ip_configuration {
+    name                          = "internal"
+    subnet_id                     = azurerm_subnet.subnet.id
+    private_ip_address_allocation = "Dynamic"
+    public_ip_address_id = azurerm_public_ip.public_ip.id 
+  }
+}
+
+# Create Network Security Group and rule
+resource "azurerm_network_security_group" "myterraformnsg" {
+    name                = "securitygroup"
+    location            = azurerm_resource_group.rg.location
+    resource_group_name = azurerm_resource_group.rg.name
+
+    security_rule {
+        name                       = "LDAP"
+        priority                   = 1002
+        direction                  = "Inbound"
+        access                     = "Allow"
+        protocol                   = "Tcp"
+        source_port_range          = "*"
+        destination_port_range     = "389"
+        source_address_prefix      = "*"
```

Comment:
If this is intended to be used ephemerally, consider using `http://ipv4.icanhazip.com` or similar to get the IP of the environment running this infra up and defaulting the inbound security rule here to only allow that IP.

Reply:
right thats an interesting idea. Looks like it wouldnt be too complex either
```
data "http" "myip" {
  url = "http://ipv4.icanhazip.com"
}
```